2021-07-25 19:28:15 下午 [Thread: main][ Class:com.zaxxer.hikari.HikariConfig >> Method: com.zaxxer.hikari.HikariConfig.setDriverClassName(HikariConfig.java:482) ]
ERROR:Failed to load driver class oracle.jdbc.driver.OracleDriver from HikariConfig class classloader sun.misc.Launcher$AppClassLoader@18b4aac2
2021-07-25 19:29:13 下午 [Thread: main][ Class:com.zaxxer.hikari.pool.HikariPool >> Method: com.zaxxer.hikari.pool.HikariPool.throwPoolInitializationException(HikariPool.java:593) ]
ERROR:HikariPool-1 - Exception during pool initialization.
java.sql.SQLRecoverableException: IO 错误: The Network Adapter could not establish the connection
	at oracle.jdbc.driver.T4CConnection.logon(T4CConnection.java:854)
	at oracle.jdbc.driver.PhysicalConnection.connect(PhysicalConnection.java:793)
	at oracle.jdbc.driver.T4CDriverExtension.getConnection(T4CDriverExtension.java:57)
	at oracle.jdbc.driver.OracleDriver.connect(OracleDriver.java:747)
	at oracle.jdbc.driver.OracleDriver.connect(OracleDriver.java:562)
	at com.zaxxer.hikari.util.DriverDataSource.getConnection(DriverDataSource.java:138)
	at com.zaxxer.hikari.pool.PoolBase.newConnection(PoolBase.java:358)
	at com.zaxxer.hikari.pool.PoolBase.newPoolEntry(PoolBase.java:206)
	at com.zaxxer.hikari.pool.HikariPool.createPoolEntry(HikariPool.java:477)
	at com.zaxxer.hikari.pool.HikariPool.checkFailFast(HikariPool.java:560)
	at com.zaxxer.hikari.pool.HikariPool.<init>(HikariPool.java:115)
	at com.zaxxer.hikari.HikariDataSource.<init>(HikariDataSource.java:81)
	at com.lyy.JDBCUtil.<clinit>(JDBCUtil.java:33)
Caused by: oracle.net.ns.NetException: The Network Adapter could not establish the connection
	at oracle.net.nt.ConnStrategy.execute(ConnStrategy.java:571)
	at oracle.net.resolver.AddrResolution.resolveAndExecute(AddrResolution.java:548)
	at oracle.net.ns.NSProtocol.establishConnection(NSProtocol.java:682)
	at oracle.net.ns.NSProtocol.connect(NSProtocol.java:309)
	at oracle.jdbc.driver.T4CConnection.connect(T4CConnection.java:1596)
	at oracle.jdbc.driver.T4CConnection.logon(T4CConnection.java:588)
	... 12 more
Caused by: java.io.IOException: Connection refused, socket connect lapse 1 ms. /0:0:0:0:0:0:0:1 1521 30000 1 true
	at oracle.net.nt.TcpNTAdapter.connect(TcpNTAdapter.java:233)
	at oracle.net.nt.ConnOption.connect(ConnOption.java:174)
	at oracle.net.nt.ConnStrategy.execute(ConnStrategy.java:510)
	... 17 more
Caused by: java.net.ConnectException: Connection refused
	at sun.nio.ch.Net.connect0(Native Method)
	at sun.nio.ch.Net.connect(Net.java:454)
	at sun.nio.ch.Net.connect(Net.java:446)
	at sun.nio.ch.SocketChannelImpl.connect(SocketChannelImpl.java:648)
	at java.nio.channels.SocketChannel.open(SocketChannel.java:189)
	at oracle.net.nt.TimeoutSocketChannel.connect(TimeoutSocketChannel.java:99)
	at oracle.net.nt.TimeoutSocketChannel.<init>(TimeoutSocketChannel.java:77)
	at oracle.net.nt.TcpNTAdapter.connect(TcpNTAdapter.java:192)
	... 19 more
